test/test_tensorflow.py

```diff
@@ -75,7 +77,11 @@ def test_horovod_allreduce_cpu(self):
         """Test on CPU that the allreduce correctly sums 1D, 2D, 3D tensors."""
         hvd.init()
         size = hvd.size()
-        dtypes = [tf.int32, tf.int64, tf.float16, tf.float32, tf.float64]
+        # MLSL supports only byte, float and double data types
```

Comment:
Do we need some sort of a check in runtime to throw an exception if the user requests unsupported compression with MLSL?

Reply:
actually, we have that kind of check. Only Allreduce needs it
The check is in mlsl_operations.cc file
`MLSL::DataType GetMLSLDataType(const std::shared_ptr<Tensor> tensor) {
  switch (tensor->dtype()) {
  case HOROVOD_UINT8:
    return MLSL::DT_BYTE;
  case HOROVOD_FLOAT32:
    return MLSL::DT_FLOAT;
  case HOROVOD_FLOAT64:
    return MLSL::DT_DOUBLE;
  default:
    throw std::logic_error("Type " + DataType_Name(tensor->dtype()) +
                           " is not supported in MLSL.");
  }
}`
This function is called when MLSL_Allreduce is executed. The reason for these checks in test_*.py files is not to let testing fail because of a bunch of types that are not supported by MLSL_Allreduce operation.
Or would you prefer seeing this check in operation.cc in ConstructResponse function?